lib/classes/Variables.js

```diff
@@ -822,6 +827,17 @@ class Variables {
       });
   }
 
+  getValueStrToBool(variableString) {
+    const groups = variableString.match(this.strToBoolRefSyntax);
+    const param = groups[1];
+    if (param === 'true') {
+      return BbPromise.resolve(true);
```

Comment:
Instead of using `BbPromise.resolve`, let's wrap whole body with `BbPromise.try`.

It'll make futer migration to async/await easier, and we don't have to deal with noise of `BbPromise.resolve` and `BbPromise.reject` in code.

---  

Good task for other PR, would be to allow resolver functions to be both `sync` and `async`. As I see it'll be purely about ensuring that non promises are handled without an error in [`promiseTracker.add`](https://github.com/serverless/serverless/blob/46aa304f5df49a651f0899e35534c56a4eeb2cdf/lib/classes/PromiseTracker.js#L39).

Also whether we need a `PromiseTracker` at all is questionable, as I see, same could be acheived with simple meoization that handles promises properly (that could be case for other PR :)